server/src/definitions/DefinitionUtils.ts

```diff
@@ -4,22 +4,82 @@
  * The whole name will be captured within a group.
  * see https://stackoverflow.com/a/51166092/3577898
  */
-const YML_ENTITY_NAME_REGEXP = new RegExp(/((\p{Letter}|[_0-9])+)/, "gu");
+const YML_ENTITY_NAME_START_REGEXP = /^((\p{Letter}|[_0-9])+)/u;
+const YML_ENTITY_NAME_END_REGEXP = /((\p{Letter}|[_0-9])+)$/u;
 
 /**
- * Find in a text the last substring that could be a valid YML entity name's substring.
+ * Find a valid token from a source text at the current offset.
+ * @param source Some text.
+ * @param offset Position of the cursor.
+ */
+export function getTokenAtPosInDoc(
+  source: string,
+  offset: number,
+): string | null {
+  if (offset == null || offset < 0 || offset > source.length) {
+    return null;
+  }
+  const documentContentToPos = source.substr(0, offset);
+
+  const firstPart = getYmlEntityNamePart(
+    documentContentToPos,
+    EntityPartPosition.END,
+  );
+
+  const documentContentFromPos = source.substr(offset);
+  const secondPart = getYmlEntityNamePart(
+    documentContentFromPos,
+    EntityPartPosition.BEGINNING,
+  );
+  if (secondPart != null && firstPart != null) {
+    // input: "variable"; position: > 0
+    // ⇒ first part and second part exists.
+    return firstPart.concat(secondPart);
+  }
+  if (secondPart != null) {
+    // input: "variable"; position: 0
+    // ⇒ no first part, but second part is a token.
+    return secondPart;
+  }
+  // No second part or nothing found.
+  // input: "variable"; position: last
+  // OR input: "variable);"; position between “)” and “;”
+  return firstPart;
+}
+
+export enum EntityPartPosition {
+  /**
+   * Search at the beginning of the input.
```

Comment:
I mean “at”

Reply:
I think the distinction is important because if you search “at” the beginning of “foobar”, you might come up with “ob”, while “from” implies that the very first characters will necessarily be comprised in the returned thingy (“fo…”). Stuff like that. Depends on the context. See what I mean.

Reply:
Damn. I thought it was the opposite :disappointed: 
I'll change it to “from”, then.

Reply:
Should it be the same in this comment?

>    * Search at the end of the input.

Reply:
I can't be sure since I have yet to fully understand what you're doing xD